node/node.go

```diff
@@ -22,108 +21,98 @@ type Node interface {
 type nexNode struct {
 	nc *nats.Conn
 
-	logger                *slog.Logger
-	agentHandshakeTimeout int
-	resourceDirectory     string
-	tags                  map[string]string
-	validIssuers          []string
-	otelOptions           OTelOptions
-	workloadOptions       []WorkloadOptions
-	hostServiceOptions    HostServiceOptions
+	// NOTE on refactor: the node struct for an operational node (state) should be
```

Comment:
I don't understand this.  Why does options need to be a separate package?  All the examples I have found using this pattern, the Options live next to the constructor.

Reply:
This is specifically so that the options can be a single thing that is passable around. It needs to be its own package because if the options are sitting on the `nexNode` struct, then we create an import loop.

Moving options shouldn't be its own PR because I literally can't pass options into the actors without options being in their own package.